app/scripts/modules/core/src/domain/IManagedEntity.ts

```diff
@@ -30,6 +30,15 @@ export interface IStatefulConstraint {
   comment?: string;
 }
 
+export interface IDependsOnConstraint {
+  type: 'depends-on';
+  currentlyPassing: boolean;
+  attributes: { environment: string };
+}
+
+// more stateless types coming soon
+export type IStatelessConstraint = IDependsOnConstraint;
+
```

Comment:
After working with this data, I'm pretty sure I think it's the wrong modeling for convenient/effective UI code usage. 

Having a totally different structure from stateful constraints means we have to jump through _tons_ of hoops for TS to be happy, do a bunch of weird casting, and constantly duplicate code paths for each type. I don't think that's been a huge deal on the backend because they're generally handled very differently, but in the UI their presentation to users is almost identical with extremely small exceptions so this pain is incredibly acute. You'll notice that in `constraintRegistry` I mostly tried to encapsulate the type differences into generic helpers that take either kind of constraint, and return whatever you need.

I'm going to see if we can move to something that looks structurally more like stateful constraints, perhaps with a `stateful` boolean flag we can use as a type discriminator to make handling a union of `IStatefulConstraint | IStatelessConstraint` productive to work with.